client/src/Pages/Login/index.js

```diff
@@ -1,11 +1,108 @@
-import React from 'react'
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import Typography from '@material-ui/core/Typography';
+import PersonIcon from '@material-ui/icons/Person';
+import { makeStyles } from '@material-ui/core';
+import Axios from 'axios';
+import Container from '../../components/Container';
+import Button from '../../components/Button';
+import InputFeild from '../../components/InputFeild';
+import validationLogin from '../../Utils/validationLogin';
+import Home from '../Home';
+
+const useStyle = makeStyles((them) => ({
+  root: {
+    margin: them.spacing(1),
+  },
+  input: {
+    width: '60%',
+  },
+  loginIcon: {
+    margin: them.spacing(2),
+  },
+}));
 
 function LoginPage() {
-    return (
-        <div>
-           <h1>Login Page</h1> 
-        </div>
-    )
+  const classes = useStyle();
+  const history = useHistory();
+  const [data, setData] = useState({});
+  const handleChange = (e) => {
+    setData((prev) => ({ ...prev, [e.target.name]: e.target.value }));
+  };
```

Comment:
You could wrap this in a `useCallback` hook to make it more efficient - then React will create the function once and reuse it, instead of recreating the function each time this component renders.
Since you have a functional update for setData, the useCallback dependencies array (the second argument to useCalback) would be empty.  I think it'd look like this:
```
const handleChange = useCallback((e) => {
    setData((prev) => ({ ...prev, [e.target.name]: e.target.value }));
  }, []);
```

Reply:
Thanks for the information